free5gc-operator/controllers/smfdeployment_controller.go

```diff
@@ -0,0 +1,648 @@
+	securityContext := &apiv1.SecurityContext{
+		Capabilities: &apiv1.Capabilities{
+			Add:  []apiv1.Capability{"NET_ADMIN"},
+			Drop: nil,
+		},
+	}
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      instanceName,
+			Namespace: namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": instanceName,
+				},
+			},
+			Template: apiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: instanceNad,
+					Labels: map[string]string{
+						"name": instanceName,
+					},
+				},
+				Spec: apiv1.PodSpec{
+					Containers: []apiv1.Container{
+						{
+							Name:            "smf",
+							Image:           smfImage,
+							ImagePullPolicy: "Always",
+							SecurityContext: securityContext,
+							Ports: []apiv1.ContainerPort{
+								{
+									Name:          "n4",
+									Protocol:      apiv1.ProtocolUDP,
+									ContainerPort: 8805,
+								},
+							},
+							Command: []string{"./smf"},
+							Args:    []string{"-c", "../config/smfcfg.yaml", "-u", "../config/uerouting.yaml"},
+							VolumeMounts: []apiv1.VolumeMount{
+								{
+									MountPath: "/free5gc/config/",
+									Name:      "smf-volume",
+								},
+							},
+							Resources: *resourceReq,
+						},
+					}, // Containers
+					DNSPolicy:     "ClusterFirst",
+					RestartPolicy: "Always",
+					Volumes: []apiv1.Volume{
+						{
+							Name: "smf-volume",
+							VolumeSource: apiv1.VolumeSource{
+								Projected: &apiv1.ProjectedVolumeSource{
+									Sources: []apiv1.VolumeProjection{
+										{
+											ConfigMap: &apiv1.ConfigMapProjection{
+												LocalObjectReference: apiv1.LocalObjectReference{
+													Name: instanceName + "-smf-configmap",
+												},
+												Items: []apiv1.KeyToPath{
+													{
+														Key:  "smfcfg.yaml",
+														Path: "smfcfg.yaml",
+													},
+													{
+														Key:  "uerouting.yaml",
+														Path: "uerouting.yaml",
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					}, // Volumes
+				}, // PodSpec
+			}, // PodTemplateSpec
+		}, // PodTemplateSpec
+	}
+	return deployment, nil
+}
+
+// Check for UERoute in SMFConfigMap yaml
+// https://github.com/s3wong/free5gc/blob/main/doc/sample-manifests/smf.yaml#L122
+// https://github.com/s3wong/free5gc/blob/main/doc/sample-manifests/smf.yaml#L272
+func free5gcSMFCreateConfigmap(logger logr.Logger, smfDeploy *workloadv1alpha1.SMFDeployment) (*apiv1.ConfigMap, error) {
+	namespace := smfDeploy.ObjectMeta.Namespace
+	instanceName := smfDeploy.ObjectMeta.Name
+
+	n4IP, err := getIPv4(smfDeploy.Spec.Interfaces, "n4")
+	if err != nil {
+		log.Log.Info("Interface N4 not found in NFDeployment Spec")
+		return nil, err
+	}
+
+	smfcfgStruct := SMFcfgStruct{}
+	smfcfgStruct.PFCP_IP = n4IP
+
+	/*
```

Comment:
Can we delete this comment?

If we want an example of what the config file looks like, perhaps we can include that. But we don't need the K8s yaml pieces, at the least.